test/jest/Netscript/StaticRamParsingCalculation.test.ts

```diff
@@ -315,8 +344,119 @@ describe("Parsing NetScript code to work out static RAM costs", function () {
-      const calculated = calculateRamUsage(code, new Map([["libTest.js" as ScriptFilePath, lib]])).cost;
+      const calculated = calculateRamUsage(
+        code,
+        filename,
+        new Map([["libTest.js" as ScriptFilePath, lib]]),
+        server,
+      ).cost;
       expectCost(calculated, GrowCost);
     });
+
+    it("Importing with a relative path - One Layer Deep", async function () {
+      const libCode = `
+          export async function testRelative(ns) {
+              await ns.hack("n00dles")
+          }
+        `;
+      const lib = new Script("test/libTest.js" as ScriptFilePath, libCode);
+      const code = `
+          import { testRelative } from "./libTest";
+
+          export async function main(ns) {
+            await testRelative(ns)
+          }
+        `;
+      const calculated = calculateRamUsage(
+        code,
+        folderFilename,
+        new Map([["test/libTest.js" as ScriptFilePath, lib]]),
+        server,
+      ).cost;
+      expectCost(calculated, HackCost);
+    });
+    it("Importing with a relative path - Two Layer Deep", async function () {
+      const libNameOne = "test/libTestOne.js" as ScriptFilePath;
+      const libNameTwo = "test/libTestTwo.js" as ScriptFilePath;
+
+      const libCodeOne = `
+          import { testRelativeAgain } from "./libTestTwo";
+          export function testRelative(ns) {
+              return testRelativeAgain(ns)
+          }
+        `;
+      const libScriptOne = new Script(libNameOne, libCodeOne);
+
+      const libCodeTwo = `
+          export function testRelativeAgain(ns) {
+              return ns.hack("n00dles")
+          }
+        `;
+      const libScriptTwo = new Script(libNameTwo, libCodeTwo);
+
+      const code = `
+          import { testRelative } from "./libTestOne";
+
+          export async function main(ns) {
+            await testRelative(ns)
+          }
+        `;
+      const calculated = calculateRamUsage(
+        code,
+        folderFilename,
+        new Map([
+          [libNameOne, libScriptOne],
+          [libNameTwo, libScriptTwo],
+        ]),
+        server,
+      ).cost;
+      expectCost(calculated, HackCost);
+    });
+    it("Importing with a relative path - possible path conflict", async function () {
+      const libNameOne = "foo/libTestOne.js" as ScriptFilePath;
+      const libNameTwo = "foo/libTestTwo.js" as ScriptFilePath;
+      const fail_libNameTwo = "test/libTestTwo.js" as ScriptFilePath;
+
+      const libCodeOne = `
+          import { testRelativeAgain } from "./libTestTwo";
+          export function testRelative(ns) {
+              return testRelativeAgain(ns)
+          }
+        `;
+      const libScriptOne = new Script(libNameOne, libCodeOne);
+
+      const libCodeTwo = `
+          export function testRelativeAgain(ns) {
+              return ns.hack("n00dles")
+          }
+        `;
+      const libScriptTwo = new Script(libNameTwo, libCodeTwo);
+
+      const fail_libCodeTwo = `
```

Comment:
I wouldn't call this ( and variables like it) "fail", because nothing about this file actually fails, does it?

Reply:
the bug i try to test here for was something i completly overlooked at first 
(resolving imports on the second/n layer based of the path of the main script and not the importing scripts path)
called it "fail" because the test would fail when it loads that file by mistake kinda a trap file

Reply:
I would call it 'incorrect" or "notLoaded" instead of "fail", then.

Reply:
but iam not that happy with the naming in the tests anyway i can take a second look on that tomorrow or if you have any names/naming scheme you like to see iam happy to implement that